subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -964,6 +964,32 @@ int ull_conn_llcp(struct ll_conn *conn, uint32_t ticks_at_expire, uint16_t lazy)
 		}
 	}
 
+#if defined(CONFIG_BT_CTLR_LE_ENC)
+	/* Run any pending procedure stored while encryption was started */
+	if (conn->llcp.encryption.pend_llcp_type != LLCP_NONE) {
```

Comment:
Move this into the previous if-clause to avoid this check being done for every connection event.

Also, use a temporary `llcp_type` and if-clause to process `event_enc_prep(conn);`  then switch using `llcp_type` assigned with `conn->llcp.encryption.pend_llcp_type`